datafusion-federation/src/analyzer.rs

```diff
@@ -20,6 +21,13 @@ impl AnalyzerRule for FederationAnalyzerRule {
     // TableScans from the same FederationProvider.
     // There 'largest sub-trees' are passed to their respective FederationProvider.optimizer.
     fn analyze(&self, plan: LogicalPlan, config: &ConfigOptions) -> Result<LogicalPlan> {
+    
+        if !contains_federated_table(&plan)? {
+            return Ok(plan);
+        }
+
+        let plan = optimize_plan(plan)?;
```

Comment:
Instead of having a single `optimize_plan`, it might make sense to make this specific to filter push down, i.e. `optimize_filter_push_down`. There are several places in `optimize_plan` that are already specific to filter push-down.

Reply:
Then the pattern becomes adding multiple rules to this top-level function

Reply:
I was also thinking about creating optimizer instance that contains list of rules to apply (array) and single config instance. Then it just walk thru all plans and apply one by one. This way we don't need to create rules + config for each query